tests/unit/test_oci.py

```diff
@@ -402,6 +402,112 @@ def test_add_layer_with_base_layer_subdirs(self, tmp_path, temp_tar_contents):
         ]
         assert temp_tar_contents == expected_tar_contents
 
+    @staticmethod
+    def _duplicate_dirs_setup(tmp_path) -> Tuple[Path, Path]:
+        """Create a filetree with an upper layer and a fake 'rootfs' structure.
+
+        layer_dir/
+          |- bin/dir1/a.txt
+          |- usr/bin/dir1/b.txt
+        rootfs/
+          |- usr/bin/
+          |- bin ----> usr/bin (symlink)
+
+        returns a tuple with (layer_dir, rootfs).
+        """
+
+        layer_dir = tmp_path / "layer_dir"
+        layer_dir.mkdir()
+
+        # The new layer has "bin/dir1/a.txt" and "usr/bin/dir1/b.txt".
+        (layer_dir / "bin/dir1").mkdir(parents=True)
+        (layer_dir / "bin/dir1/a.txt").touch()
+        (layer_dir / "usr/bin/dir1").mkdir(parents=True)
+        (layer_dir / "usr/bin/dir1/b.txt").touch()
+
+        rootfs_dir = tmp_path / "rootfs"
+        rootfs_dir.mkdir()
+        # In the base layer "bin" is a symlink to "usr/bin"
+        (rootfs_dir / "usr/bin").mkdir(parents=True)
+        os.symlink("usr/bin", rootfs_dir / "bin")
```

Comment:
fyi, `pathlib.Path` has the amazing `symlink_to`

Reply:
thanks! I have a strong memory of looking at `symlink_to` and not being able to use it for "some reason", but I updated the code now and it seems to work fine? weird